src/Spec/Retrieve.purs

```diff
@@ -0,0 +1,115 @@
+-- | A module with functions for retriving particular fields from a form
+module Formless.Spec.Retrieve where
+
+import Prelude
+
+import Data.Either (Either, either, hush)
+import Data.Lens (preview, view)
+import Data.Maybe (Maybe(..))
+import Data.Newtype (class Newtype, unwrap)
+import Data.Symbol (class IsSymbol, SProxy)
+import Formless.Spec (FormField(..), FormFieldRow, _Error, _Field, _Input, _Output, _Result, _Touched)
+import Heterogeneous.Mapping (class HMap, class Mapping, hmap)
+import Prim.Row as Row
+
+----------
+-- Fields
+
+-- | A type representing a function to produce a value from a record of
+-- | form fields given a particular symbol. The result of `view` from
+-- | Data.Lens applied with a particular lens to the form.
+type FormFieldGet e i o x =
+  ∀ sym form fields t0
+   . IsSymbol sym
+  => Newtype (form Record FormField) (Record fields)
+  => Row.Cons sym (FormField e i o) t0 fields
+  => SProxy sym
+  -> form Record FormField
+  -> x
+
+-- | Given a form, get the field at the specified symbol
+getField :: ∀ e i o. FormFieldGet e i o (Record (FormFieldRow e i o))
+getField sym = view (_Field sym)
+
+-- | Given a form, get the input at the specified symbol
+getInput :: ∀ e i o. FormFieldGet e i o i
+getInput sym = view (_Input sym)
+
+-- | Given a form, get the touched field at the specified symbol
+getTouched :: ∀ e i o. FormFieldGet e i o Boolean
+getTouched sym = view (_Touched sym)
+
+-- | Given a form, get the result at the specified symbol
+getResult :: ∀ e i o. FormFieldGet e i o (Maybe (Either e o))
+getResult sym = view (_Result sym)
+
+-- | Given a form, get the error (if it exists) at the specified symbol
+getError :: ∀ e i o. FormFieldGet e i o (Maybe e)
+getError sym = preview (_Error sym)
+
+-- | Given a form, get the output (if it exists) at the specified symbol
+getOutput :: ∀ e i o. FormFieldGet e i o (Maybe o)
+getOutput sym = preview (_Output sym)
+
+
+----------
+-- Summary functions
+
+-- | A type representing retrieving all of a particular field with the field's
+-- | constructor name. For internal use.
+type GetAll f =
```

Comment:
@dariooddenino Your version of this is prettier and the type signatures are more immediately obvious in explaining what the function does (`getInputs :: FormFieldGetFields "input"`). However, I'm trying to minimize the number of constraints (especially `RowToList` ones) wherever possible for compile-time performance reasons. This version is less nice, but has fewer constraints.